src/net_processing.cpp

```diff
@@ -1975,6 +1977,51 @@ static bool ProcessGetCFilters(CNode* pfrom, CDataStream& vRecv, const CChainPar
     return true;
 }
 
+static bool ProcessGetCFHeaders(CNode* pfrom, CDataStream& vRecv, const CChainParams& chain_params,
+                                CConnman* connman)
+{
+    uint8_t filter_type_ser;
+    uint32_t start_height;
+    uint256 stop_hash;
+
+    vRecv >> filter_type_ser >> start_height >> stop_hash;
+
+    BlockFilterType filter_type = static_cast<BlockFilterType>(filter_type_ser);
+
+    const CBlockIndex* stop_index;
+    BlockFilterIndex* filter_index;
+    if (!PrepareBlockFilterRequest(pfrom, chain_params, filter_type, start_height, stop_hash,
+                                   MAX_GETCFHEADERS_SIZE, stop_index, filter_index)) {
+        // Return true because the issue with the invalid request has already been logged.
+        return true;
+    }
+
+    uint256 prev_header;
+    if (start_height > 0) {
```

Comment:
07dd762

What the purpose of checking filters header availability before start of the range ? It seems like zeal on the bip

Reply:
What do you mean? Like fetching the filter at `start_height - 1`? The purpose is that the one filter header before the start of the range is required to compute the filter headers for all filters in the range given their hashes (it's a hash chain).